base/dcim.yml

```diff
@@ -97,75 +97,194 @@ generics:
         kind: Attribute
         order_weight: 1500
 
+  - name: Endpoint
+    namespace: Dcim
+    description: Generic Endpoint to receive a connector.
+    include_in_menu: false
+    relationships:
+      - name: connector
+        peer: DcimConnector
+        optional: true
+        cardinality: one
+        order_weight: 1500
+        kind: Attribute
+
+  - name: Connector
+    namespace: Dcim
+    description: Generic Connector to link two endpoints together.
+    include_in_menu: false
+    relationships:
+      - name: connected_endpoints
+        peer: DcimEndpoint
+        optional: true
+        cardinality: many
+        order_weight: 1500
+        kind: Generic
+
+    # Interfaces
   - name: Interface
     namespace: Dcim
-    description: Generic Network Interface.
-    label: Interface
-    icon: mdi:ethernet
-    menu_placement: DcimGenericDevice
-    include_in_menu: true
+    description: "Generic Network Interface"
+    label: "Interface"
+    include_in_menu: false
     display_labels:
       - name__value
     order_by:
       - device__name__value
       - name__value
     uniqueness_constraints:
-      - [device, name__value]
+      - ["device", "name__value"]
+    human_friendly_id:
+      - "device__name__value"
+      - "name__value"
     attributes:
       - name: name
         kind: Text
+        description: "Name of the interface"
         order_weight: 1000
       - name: description
         kind: Text
         optional: true
+        description: "A brief description of the interface"
         order_weight: 1100
-      - name: speed
-        kind: Number
-        order_weight: 1400
       - name: mtu
         label: MTU
-        default_value: 1500
         kind: Number
-        order_weight: 1500
-      - name: enabled
-        kind: Boolean
-        default_value: true
+        default_value: 1514
+        order_weight: 1300
+      - name: status
```

Comment:
As we can't edit dropdown on generic I don't know if we shouldn't put it in the actual node ... (even tho it's going to be duplicated ...)

Reply:
you can override in a node the dropdown by recreating the attribute with the keyword "inherited: true" to redefine the options.

I would let it in the generic, and people can override it afterwards if they need other options.